openwillis/measures/audio/speech_phonation.py

```diff
@@ -0,0 +1,214 @@
+
+    Parameters:
+    ...........
+    filepath : str
+        Path to the input audio file.
+    transcript_json : json
+        Speech transcription json response.
+    speaker_label : str
+        Speaker label.
+
+    Returns:
+    ...........
+    phonation_dict : pandas.DataFrame
+        A pandas dataframe containing the phonation information.
+
+    ------------------------------------------------------------------------------------------------------
+    """
+    phonation_dict = {}
+
+    try:
+        audio_signal = AudioSegment.from_file(file = filepath, format = "wav")
+        if is_whisper_transcribe(transcript_json):
+            speaker_df, _ = sutil.whisperx_to_dataframe(transcript_json)
+        elif is_amazon_transcribe(transcript_json):
+            speaker_df, _ = sutil.transcribe_response_to_dataframe(transcript_json)
+        else:
+            speaker_df = sutil.vosk_to_dataframe(transcript_json)
+
+        if speaker_label:
+            speaker_df = speaker_df[speaker_df['speaker_label']==speaker_label]
+            if speaker_df.empty:
+                raise ValueError(f"Speaker label {speaker_label} not found in the transcript.")
+        elif speaker_df['speaker_label'].nunique() > 1:
+            raise ValueError("Multiple speakers found in the transcript. Please provide a speaker label.")
+
+        phonation_df = putil.extract_phonation(speaker_df)
+        if not phonation_df.empty:
+            phonation_dict = putil.segment_phonations(audio_signal, phonation_df)
+
+    except Exception as e:
+        logger.error(f'Error phonation extraction: {e} & File: {filepath}')
+
+    return phonation_dict
+
+def clean_acoustic_df(df, file, duration, measures):
+    """
+    ------------------------------------------------------------------------------------------------------
+
+    This function cleans the acoustic dataframe.
+
+    Parameters:
+    ...........
+    df : pandas.DataFrame
+        input dataframe
+    file : str
+        file name
+    duration : float
+        duration of the audio signal in seconds
+    measures : dict
+        column names configuration
+
+    Returns:
+    ...........
+    pandas.DataFrame
+        cleaned dataframe
+
+    ------------------------------------------------------------------------------------------------------
+    """
+    df[measures['phonation_type']] = file.split('_')[-1][0]
+    df = df[[measures['phonation_type']] + [col for col in df.columns if col != measures['phonation_type']]]
+    df[measures['duration']] = duration
+
+    # remove unused columns - pause related measures
+    cols_to_drop = [measures['spir'], measures['pause_meddur'], measures['pause_maddur'], measures['silence_ratio']]
+    df = df.drop(columns=[col for col in cols_to_drop if col in df.columns], errors='ignore')
+
+    return df
+
+def phonations_acoustics(audio_path, transcript_json, speaker_label=''):
```

Comment:
Just checking : this function should work for en language only. right? Should we add a language check or not?

Reply:
That's right yes, this works only for english. We cannot add a language check because we dont have language as a parameter, nor should we add it I guess just to check if it's en. Users are supposed to only use this function if their data are in English language